src/Diagrams/Backend/SVG.hs

```diff
@@ -277,6 +284,18 @@ svgDefinitions f opts =
 idPrefix :: SVGFloat n => Lens' (Options SVG V2 n) T.Text
 idPrefix f opts = f (_idPrefix opts) <&> \i -> opts { _idPrefix = i }
 
+-- | Lens onto the svgAttributes field of the svg options. This field
+--   is provided to supply SVG attributes to the entire diagram.
+svgAttributes :: SVGFloat n => Lens' (Options SVG V2 n) [Attribute]
+svgAttributes f opts =
+  f (_svgAttributes opts) <&> \ds -> opts { _svgAttributes = ds }
+
+-- | Lens onto the generateDoctype field of the svg options. Set
+--   to False if you don't want a doctype tag included in the output.
+generateDoctype :: SVGFloat n => Lens' (Options SVG V2 n) Bool
+generateDoctype f opts =
+  f (_generateDoctype opts) <&> \ds -> opts { _generateDoctype = ds }
```

Comment:
Adding some use examples is a good idea, I'll see what i can work up

On Wed, Sep 9, 2015 at 4:45 PM Brent Yorgey <notifications@github.com>
wrote:

> In src/Diagrams/Backend/SVG.hs
> <https://github.com/diagrams/diagrams-svg/pull/85#discussion_r39093830>:
>
> > @@ -277,6 +284,18 @@ svgDefinitions f opts =
> >  idPrefix :: SVGFloat n => Lens' (Options SVG V2 n) T.Text
> >  idPrefix f opts = f (_idPrefix opts) <&> \i -> opts { _idPrefix = i }
> >
> > +-- | Lens onto the svgAttributes field of the svg options. This field
> > +--   is provided to supply SVG attributes to the entire diagram.
> > +svgAttributes :: SVGFloat n => Lens' (Options SVG V2 n) [Attribute]
> > +svgAttributes f opts =
> > +  f (_svgAttributes opts) <&> \ds -> opts { _svgAttributes = ds }
> > +
> > +-- | Lens onto the generateDoctype field of the svg options. Set
> > +--   to False if you don't want a doctype tag included in the output.
> > +generateDoctype :: SVGFloat n => Lens' (Options SVG V2 n) Bool
> > +generateDoctype f opts =
> > +  f (_generateDoctype opts) <&> \ds -> opts { _generateDoctype = ds }
>
> Ah, I see. Never mind then.
> … <#msg-f:1511869948096927201_>
> On 3:24pm, Wed, Sep 9, 2015 Chris <notifications@github.com> wrote: In
> src/Diagrams/Backend/SVG.hs <
> https://github.com/diagrams/diagrams-svg/pull/85#discussion_r39091334>: >
> @@ -277,6 +284,18 @@ svgDefinitions f opts = > idPrefix :: SVGFloat n =>
> Lens' (Options SVG V2 n) T.Text > idPrefix f opts = f (_idPrefix opts) <&>
> \i -> opts { _idPrefix = i } > > +-- | Lens onto the svgAttributes field of
> the svg options. This field > +-- is provided to supply SVG attributes to
> the entire diagram. > +svgAttributes :: SVGFloat n => Lens' (Options SVG V2
> n) [Attribute] > +svgAttributes f opts = > + f (_svgAttributes opts) <&>
> \ds -> opts { _svgAttributes = ds } > + > +-- | Lens onto the
> generateDoctype field of the svg options. Set > +-- to False if you don't
> want a doctype tag included in the output. > +generateDoctype :: SVGFloat n
> => Lens' (Options SVG V2 n) Bool > +generateDoctype f opts = > + f
> (_generateDoctype opts) <&> \ds -> opts { _generateDoctype = ds } I'm not
> sure you can with makeLenses because it's a data instance. You should be
> able to generate it with declareLenses but then you lose the record syntax
> versions (_size etc.), which I think some people still like to use (I
> personally don't mind losing them). — Reply to this email directly or view
> it on GitHub <
> https://github.com/diagrams/diagrams-svg/pull/85/files#r39091334>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diagrams/diagrams-svg/pull/85/files#r39093830>.
>